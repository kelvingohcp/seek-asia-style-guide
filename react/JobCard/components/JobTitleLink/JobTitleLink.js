import React from 'react';
import PropTypes from 'prop-types';
import styles from './JobTitleLink.less';
import classnames from 'classnames';
import { getParts } from '../../jobCardHelper.js';
import { Text } from 'seek-asia-style-guide/react';
import defaultLink from '../Link/Link';

const JobTitleLinkPropTypes = PropTypes.shape({
  jobTitle: PropTypes.string,
  jobUrl: PropTypes.string
});

const JobTitleLink = ({ keyword, job: { jobTitle, jobUrl }, LinkComponent = defaultLink, viewed }) => {
  let title = (<Text whistling strong className={styles.jobTitle}>{jobTitle}</Text>);
  const keywordParts = getParts(jobTitle, keyword);

  if (keywordParts) {
    title = (
      <span>
        {
          keywordParts.map((part, index) => {
            return (
              <Text
                strong
                whistling
                className={classnames(styles.jobTitle, { [styles.titleKeyword]: part.highlight })}
                key={index}>
                {part.text}
              </Text>
            );
          })
        }
      </span>
    );
  }

  return <LinkComponent linkType="job-title-link" link={jobUrl} className={classnames({ [styles.jobTitleLink]: !viewed, [styles.jobTitleLinkVisited]: viewed })} rel="noopener noreferrer">{title}</LinkComponent>;
};

JobTitleLink.propTypes = {
  keyword: PropTypes.string,
  showJobSplit: PropTypes.bool,
  isDesktop: PropTypes.bool,
  job: JobTitleLinkPropTypes.isRequired,
  LinkComponent: PropTypes.func,
  viewed: PropTypes.bool
};

export default JobTitleLink;
