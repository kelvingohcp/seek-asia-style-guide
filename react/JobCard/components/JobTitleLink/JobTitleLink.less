@import (reference) "~seek-asia-style-guide/theme";
@job-card-text-space: 0;

.jobTitle {
  padding-bottom: @job-card-text-space;
  display: inline;
  word-wrap: break-word;
}

.jobTitleLink {
  .link(@actionBlue, true);
  text-decoration: none;
  pointer-events: auto;

  .titleKeyword {
    color: darken(@actionBlue, 10%);
  }
}

.jobTitleLinkVisited,
.jobTitleLink:visited {
  .link(@saVisitedLink, true);

  .titleKeyword {
    color: darken(@saVisitedLink, 10%);
  }
}
