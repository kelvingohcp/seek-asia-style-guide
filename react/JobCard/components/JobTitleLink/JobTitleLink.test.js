import React from 'react';
import { shallow } from 'enzyme';

import JobTitleLink from './JobTitleLink';

const defaultJob = {
  jobTitle: 'Software Engineer',
  jobUrl: '/jobCard'
};

describe('JobCard - JobTitleLink', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<JobTitleLink job={defaultJob} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should not bold if the keyword are not match', () => {
    const wrapper = shallow(<JobTitleLink job={defaultJob} keyword="Programmer" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should bold keyword', () => {
    const wrapper = shallow(<JobTitleLink job={defaultJob} keyword="Soft Eng" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render LinkComponent when it is provided', () => {
    const wrapper = shallow(
      <JobTitleLink
        job={defaultJob}
        keyword="Engineer"
        LinkComponent={({ link, children, ...restProps }) => (
          <span {...restProps}>{children}</span>
        )}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should change className for viewed job', () => {
    const wrapper = shallow(<JobTitleLink job={defaultJob} viewed />);
    expect(wrapper).toMatchSnapshot();
  });
});
