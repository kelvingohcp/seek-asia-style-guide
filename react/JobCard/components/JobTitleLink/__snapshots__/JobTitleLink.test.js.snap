// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard - JobTitleLink should bold keyword 1`] = `
<Link
  className="jobTitleLink"
  link="/jobCard"
  rel="noopener noreferrer"
>
  <Text
    baseline={true}
    className="jobTitle"
    strong={true}
    waving={true}
  >
    Software Engineer
  </Text>
</Link>
`;

exports[`JobCard - JobTitleLink should pass through viewed prop 1`] = `
<Link
  className="jobTitleLinkVisited"
  link="/jobCard"
  rel="noopener noreferrer"
  viewed={true}
>
  <Text
    baseline={true}
    className="jobTitle"
    strong={true}
    waving={true}
  >
    Software Engineer
  </Text>
</Link>
`;

exports[`JobCard - JobTitleLink should render LinkComponent when it is provided 1`] = `
<LinkComponent
  className="jobTitleLink"
  link="/jobCard"
  rel="noopener noreferrer"
>
  <Text
    baseline={true}
    className="jobTitle"
    strong={true}
    waving={true}
  >
    Software Engineer
  </Text>
</LinkComponent>
`;

exports[`JobCard - JobTitleLink should render correctly 1`] = `
<Link
  className="jobTitleLink"
  link="/jobCard"
  rel="noopener noreferrer"
>
  <Text
    baseline={true}
    className="jobTitle"
    strong={true}
    waving={true}
  >
    Software Engineer
  </Text>
</Link>
`;
